File: benchmarks/cases/listops/listops.dart
// List operations — build arrays, then sum + count below threshold.
import 'dart:io';
import 'dart:math';

const int minstd = 16807;
const int modulus = 2147483647;
const int bigMod = 1000000007;
const int n = 4000;

int r1(int x) => (x * minstd) % modulus;

int hashAt(int s, int i) => r1(r1(s + i));

int readSeed() {
  final line = stdin.readLineSync();
  int m = 0;
  if (line != null) {
    m = int.tryParse(line.trim()) ?? 0;
  }
  if (m == 0) {
    return 1;
  }
  final rng = Random.secure();
  final val = (rng.nextInt(1 << 31) << 31) | rng.nextInt(1 << 31);
  return (val % 2147483646) + 1;
}

void main() {
  final seed = readSeed();

  int acc = 0;
  for (int t = 0; t < 8; t++) {
    final s = seed + t * 131;
    final xs = List<int>.filled(n, 0);
    for (int i = 0; i < n; i++) {
      xs[i] = hashAt(s, i) % 1000;
    }
    int sum = 0, below = 0;
    for (int i = 0; i < n; i++) {
      sum += xs[i];
      if (xs[i] < 500) {
        below++;
      }
    }
    acc = (acc + sum + below) % bigMod;
  }

  print(acc);
}

File: benchmarks/cases/listops/listops.cs
// List operations — build arrays, then sum + count below threshold.
const long MINSTD = 16807;
const long MODULUS = 2147483647;
const long BIG_MOD = 1000000007;
const int N = 4000;

long seed = ReadSeed();

long acc = 0;
for (long t = 0; t < 8; t++)
{
    long s = seed + t * 131;
    long[] xs = new long[N];
    for (long i = 0; i < N; i++)
    {
        xs[i] = HashAt(s, i) % 1000;
    }
    long sum = 0, below = 0;
    for (long i = 0; i < N; i++)
    {
        sum += xs[i];
        if (xs[i] < 500)
        {
            below++;
        }
    }
    acc = (acc + sum + below) % BIG_MOD;
}

System.Console.WriteLine(acc);

static long R1(long x) => (x * MINSTD) % MODULUS;

static long HashAt(long s, long i) => R1(R1(s + i));

static long ReadSeed()
{
    string line = System.Console.ReadLine();
    long m = 0;
    if (line != null)
    {
        long parsed;
        if (long.TryParse(line.Trim(), out parsed))
        {
            m = parsed;
        }
    }
    if (m == 0)
    {
        return 1;
    }
    byte[] buf = new byte[8];
    System.Security.Cryptography.RandomNumberGenerator.Fill(buf);
    ulong val = System.BitConverter.ToUInt64(buf, 0);
    return (long)(val % 2147483646UL) + 1;
}
